stubs/urllib3/urllib3/connectionpool.pyi

```diff
@@ -29,48 +34,52 @@ Retry = retry.Retry
 Timeout = timeout.Timeout
 get_host = url.get_host
 
+_Timeout: TypeAlias = Timeout | float
+_Retries: TypeAlias = Retry | bool | int
+
 xrange: Any
-log: Any
+log: Logger
 
 class ConnectionPool:
-    scheme: Any
-    QueueCls: Any
-    host: Any
-    port: Any
-    def __init__(self, host, port=...) -> None: ...
-    def __enter__(self): ...
-    def __exit__(self, exc_type, exc_val, exc_tb): ...
-    def close(self): ...
+    scheme: ClassVar[str | None]
+    QueueCls: ClassVar[Any]
+    host: str
+    port: int | None
+    def __init__(self, host: str, port: int | None = ...) -> None: ...
+    def __enter__(self: Self) -> Self: ...
+    def __exit__(self, exc_type, exc_val, exc_tb) -> bool: ...
+    def close(self) -> None: ...
 
 class HTTPConnectionPool(ConnectionPool, RequestMethods):
-    scheme: Any
-    ConnectionCls: Any
-    strict: Any
-    timeout: Any
-    retries: Any
-    pool: Any
-    block: Any
-    proxy: Any
-    proxy_headers: Any
-    num_connections: Any
-    num_requests: Any
+    scheme: ClassVar[str]
+    ConnectionCls: ClassVar[type[HTTPConnection]]
+    ResponseCls: ClassVar[type[HTTPResponse]]
+    strict: bool
+    timeout: _Timeout
+    retries: _Retries | None
+    pool: LifoQueue[Any] | None
+    block: bool
+    proxy: Url | None
+    proxy_headers: Mapping[str, str] | None
```

Comment:
If `None` is passed into the `__init__` parameter, it's set to an empty dictionary: https://github.com/urllib3/urllib3/blob/aa3def7d242525e6e854991247c4b68583d15135/src/urllib3/connectionpool.py#L202

```suggestion
    proxy_headers: Mapping[str, str]
```